Source/WebKit/UIProcess/API/glib/WebKitDownload.cpp

```diff
@@ -73,12 +73,21 @@ enum {
 static GParamSpec* sObjProperties[N_PROPERTIES] = { nullptr, };
 
 struct _WebKitDownloadPrivate {
+    ~_WebKitDownloadPrivate()
+    {
+        if (decideDestinationCallback) {
+            g_critical("Bug: application handled WebKitDownload::decide-destination but failed to call webkit_download_set_destination() before the WebKitDownload was destroyed");
+            (*decideDestinationCallback)(AllowOverwrite::No, { });
+        }
```

Comment:
CompletionHandler already asserts when destroyed without being invoked, I'm not sure we need this.

Reply:
Well it's optional to check this here, but (a) a g_critical() matches the precondition assertions we do using g_return_if_fail(), and (b) CompletionHandler asserts are only enabled in debug builds anyway, so app developers will never see them.